attachment_s3/models/ir_attachment.py

```diff
@@ -227,10 +239,16 @@ def _file_read_s3(self, fname, bin_size=False):
                 "error reading attachment '%s' from object storage", fname
             )
             return ''
-        filekey = bucket.get_key(s3uri.item())
-        if filekey:
-            read = base64.b64encode(filekey.get_contents_as_string())
-        else:
+        try:
+            key = s3uri.item()
+            bucket.meta.client.head_object(
+                Bucket=bucket.name,  Key=key
+            )
+            res = StringIO.StringIO()
```

Comment:
shouldn't close this?

Reply:
@simahawk I tried with my last fixup but it didn't work.
Not supported in Python 2: https://bugs.python.org/issue1286

```python
[...]
  File "/odoo/external-src/odoo-cloud-platform/attachment_s3/models/ir_attachment.py", line 246, in _file_read_s3
    with StringIO() as res:
AttributeError: __exit__
```

Reply:
you have to use `contextlib.closing` :wink: 
```
from contextlib import closing
[...]
with closing(...):
    # do it
[...]
```
or simply close it "manually" once you are done